configuration/cdap-site.xml

```diff
@@ -138,283 +169,257 @@
     <description>
-      Max connection backlog of CDAP Master
+      ZooKeeper quorum string; specifies the ZooKeeper host:port; substitute the quorum
+      (FQDN1:2181,FQDN2:2181,...) for the components shown here
     </description>
   </property>
 
   <property>
-    <name>app.exec.threads</name>
-    <value>20</value>
+    <name>zookeeper.session.timeout.millis</name>
+    <value>40000</value>
     <description>
-      Number of Netty service executor threads
+      ZooKeeper session timeout in milliseconds
     </description>
   </property>
 
-  <property>
-    <name>app.boss.threads</name>
-    <value>1</value>
-    <description>
-      Number of Netty service boss threads
-    </description>
-  </property>
+  <!-- Global Configuration -->
 
   <property>
-    <name>app.worker.threads</name>
-    <value>10</value>
+    <name>dataset.unchecked.upgrade</name>
+    <value>false</value>
     <description>
-      Number of Netty service worker threads
+      By default, any changes made to existing datasets are not deployed
+      when an app is redeployed; setting this value to true allows the
+      dataset changes to be deployed upon app redeployment.
     </description>
-  </property>
-
-  <!-- YARN Services Parameters -->
-
-  <property>
-    <name>master.service.num.cores</name>
-    <value>2</value>
-    <description>Number of cores for Master Service instance</description>
-    <display-name>CDAP Master YARN CPU Cores</display-name>
     <value-attributes>
-      <type>int</type>
-      <minimum>1</minimum>
-      <maximum>4</maximum>
-      <increment-step>1</increment-step>
+      <type>boolean</type>
       <overridable>false</overridable>
     </value-attributes>
   </property>
 
   <property>
-    <name>master.service.memory.mb</name>
-    <value>512</value>
+    <name>enable.unrecoverable.reset</name>
+    <value>false</value>
     <description>
-      Size of memory in megabytes for Master Service instance
+      Determines if resetting CDAP should be enabled. **WARNING: Enabling
+      this option makes it possible to delete all applications and data; NO
+      RECOVERY IS POSSIBLE!**
     </description>
-    <display-name>CDAP Master YARN Memory</display-name>
     <value-attributes>
-      <type>int</type>
-      <minimum>0</minimum>
-      <maximum>268435456</maximum>
-      <unit>MB</unit>
-      <increment-step>256</increment-step>
+      <type>boolean</type>
       <overridable>false</overridable>
     </value-attributes>
   </property>
 
+  <!-- Applications Configuration -->
+
   <property>
-    <name>master.service.max.instances</name>
-    <value>5</value>
-    <description>Maximum number of Master Service instances</description>
+    <name>app.artifact.dir</name>
+    <value>/opt/cdap/master/artifacts</value>
+    <description>
+      Semicolon-separated list of local directories scanned for system artifacts
+      to add to the artifact repository
+    </description>
   </property>
 
```

Comment:
I guess I should have mentioned that. Those properties are *really* useless on a distributed cluster. They're generally used for switching between `0.0.0.0` and `127.0.0.1` in SDK... I cannot think of a case where we would want a customer touching this, since they would have no idea what node will get the container ahead of time.

I kept the bind addresses for the actual daemons with listening ports... router, kafka, ui